Realm/ObjectServerTests/RLMSyncTestCase.m

```diff
@@ -113,44 +183,67 @@ + (void)setUp {
     }
 }
 
-- (void)setUp {
-    [super setUp];
-    if (!self.isParent) {
-        // Don't start the sync server if not the originating process.
+- (void)lazilyInitializeObjectServer {
+    if (!self.isParent || s_task) {
         return;
     }
-    // FIXME: we need a more robust way of waiting till a test's server process has completed cleaning
-    // up before starting another test.
-    sleep(1);
-    [RLMSyncManager sharedManager].logLevel = RLMSyncLogLevelOff;
-    self.task = [[NSTask alloc] init];
-    self.task.currentDirectoryPath = [[RLMSyncTestCase rootRealmCocoaURL] path];
-    self.task.launchPath = @"/bin/sh";
-    self.task.arguments = @[@"build.sh", @"start-object-server"];
+    NSTask *task = [[NSTask alloc] init];
+    task.currentDirectoryPath = [[RLMSyncTestCase rootRealmCocoaURL] path];
+    task.launchPath = @"/bin/sh";
+    task.arguments = @[@"build.sh", @"start-object-server"];
     dispatch_semaphore_t sema = dispatch_semaphore_create(0);
-    self.task.standardOutput = [NSPipe pipe];
-    [[self.task.standardOutput fileHandleForReading] setReadabilityHandler:^(NSFileHandle *file) {
+    task.standardOutput = [NSPipe pipe];
+    [[task.standardOutput fileHandleForReading] setReadabilityHandler:^(NSFileHandle *file) {
         NSData *data = [file availableData];
         NSString *output = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
         if ([output containsString:@"Received: IDENT"]) {
             dispatch_semaphore_signal(sema);
         }
     }];
-    [self.task launch];
-    dispatch_semaphore_wait(sema, DISPATCH_TIME_FOREVER);
+    s_task = task;
+    [task launch];
+    const NSTimeInterval timeout = 60;
+    BOOL wait_result = dispatch_semaphore_wait(sema, dispatch_time(DISPATCH_TIME_NOW, (int64_t)(timeout * NSEC_PER_SEC))) == 0;
+    if (!wait_result) {
+        s_task = nil;
+        XCTFail(@"Server did not start up within the allotted timeout interval.");
+    }
 }
 
-- (void)tearDown {
++ (void)tearDown {
+    [s_task terminate];
+    [super tearDown];
+}
+
++ (void)runResetObjectServer {
```

Comment:
Why is this necessary now that each test runs in its own "isolated" environment? i.e. different Realms and user accounts.